vue/src/vueTextMask.js

```diff
@@ -0,0 +1,49 @@
+import createTextMaskInputElement from '../../core/src/createTextMaskInputElement.js'
+
+/*
+ * Vue Directive - Text Masking with Vanilla Text Mask
+ * https://github.com/text-mask/text-mask/tree/master/vanilla
+ *
+ * This directive binds input masking via Text Mask. Once the element is
+ * destroyed it will be unbinded. Pass an options object, same as you would
+ * normally, as a parameter on the element being bound.
+ */
+
+/**
+ * Initialize variable to store directive's `this` so that we can use its
+ * context in inputHandler called via event listener
+ */
+let instance
+
+// Register the input handler to watch for updates
+const inputHandler = ({target: {value}}) => {
+  return instance.textMaskInputElement.update(value)
+}
+
+export default {
+  textMaskInputElement: 0,
```

Comment:
You can initialize this to `null`. And inside the `bind` method, you could do something like:

```js
const options = this.params.maskOptions || {}
options.inputElement = this.el
this.textMaskInputElement = createTextMaskInputElement(options)

// We need to bind the input handler to the context of this directive so 
// that when it's called by the event listener, it 
// will be able to see `this.textMaskInputElement`
this.inputHandler = this.inputHandler.bind(this)
this.el.addEventListener('input', this.inputHandler)
```